dvc/remote/gdrive.py

```diff
@@ -155,51 +160,89 @@ def _validate_config(self):
             or not self._service_account_p12_file_path
         ):
             raise DvcException(
-                "To use service account please specify {}, {} and "
-                "{} in DVC config. Learn more at "
-                "{}.".format(
-                    "gdrive_service_account_email",
-                    "gdrive_service_account_p12_file_path",
-                    "gdrive_service_account_user_email (optional)",
-                    format_link("https://man.dvc.org/remote/modify"),
+                "To use service account set 'gdrive_service_account_email',\n"
+                "'gdrive_service_account_p12_file_path', and optionally "
+                "'gdrive_service_account_user_email'\nin DVC config. "
+                "Learn more at {}".format(
+                    format_link("https://man.dvc.org/remote/modify")
                 )
             )
 
         # Validate OAuth 2.0 Client ID configuration
         if not self._use_service_account:
             if bool(self._client_id) != bool(self._client_secret):
                 raise DvcException(
-                    "Please specify Google Drive's client id and secret in "
-                    "DVC config or omit both to use defaults. Learn more at "
-                    "{}.".format(
+                    "Please specify GDrive's client id and secret in "
+                    "DVC config or omit both to use defaults.\n"
+                    "Learn more at {}".format(
+                        format_link("https://man.dvc.org/remote/modify")
+                    )
+                )
+
+    @cached_property
+    def _credentials_location(self):
+        if os.getenv(GDriveRemote.GDRIVE_CREDENTIALS_DATA):
+            return GDriveRemote.GDRIVE_CREDENTIALS_DATA
+        if os.path.exists(self._gdrive_user_credentials_path):
+            return self._gdrive_user_credentials_path
+        return None
+
+    # Detects discrepancy in DVC config and cached credentials file.
+    # Usually happens when a second remote is added and it is using
+    # the same credentials default file. Or when someones decides to change
+    # DVC config client id or secret but forgets to remove the cached
+    # credentials file.
+    @staticmethod
+    def _validate_credentials(auth, settings):
+        if not os.getenv(GDriveRemote.GDRIVE_CREDENTIALS_DATA):
+            if (
+                settings["client_config"]["client_id"]
+                != auth.credentials.client_id
+                or settings["client_config"]["client_secret"]
+                != auth.credentials.client_secret
+            ):
+                logger.warning(
+                    "Client secret and id from the DVC config do not match "
+                    "actual ones used\nto access the remote storage. Do you "
+                    "use multiple GDrive remotes and forgot to\nset "
+                    "'gdrive_user_credentials_file' for one or more of them? "
+                    "Learn more at\n{}.\n".format(
```

Comment:
```suggestion
                    "Client ID and secret configured do not match the "
                    "actual ones used\nto access the remote. Do you "
                    "use multiple GDrive remotes and forgot to\nset "
                    "`gdrive_user_credentials_file` for one or more of them? "
                    "Learn more at\n{}.\n".format(
```
> Needs rewrap

But I'm not sure I understand this part: `... do not match the actual ones used access the remote.`

Reply:
Same insisting suggestion about upper case ID, back quotes, and also I don't know if "from the DVC config" is a phrase that's easy to understand, but not sure how else to say it. Just "configured" may not be specific enough.